### PR TITLE
WIP: add contributing.md file

### PR DESCRIPTION
**Pull request by ratanparai**
Work in progress. 

Will merge it after completing the guide

### CODE REVIEW

**Review by mhmohona — APPROVED** [review]
I think this is okay

**Review by cseshohag — APPROVED** [review]
Great;
Thank you @ratanparai Bhai for a good contribution guide line.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
+# How to contribute
+
+Before going deep into the steps please remember you **DO NOT NEED** to follow the remote name we used. You can use anything you want. You need to change your command depending on the name you choose.
```
Comment (mahedee) [inline CONTRIBUTING.md]: Please add overview of workflow diagram

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+- **upstream** - Remote name of `techcomdbd`'s repository.
```
Comment (mahedee) [inline CONTRIBUTING.md]: What is upstream and origin. please explain before.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+- **origin** - Remote name of your forked repository.
+
+## Steps for doing a pull request
+
+1. Fork the repository
+
+2. Clone the forked repository to your computer
```
Comment (mahedee) [inline CONTRIBUTING.md]: Add like.. Clone the forked repository to your computer using the following command.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+git clone git@github.com:ratanparai/first-contributions.git
+```
+
+3. Now `cd` into the cloned folder-
+
+```
+cd first-contributions
+```
+
+3. add upstream remote-
```
Comment (mahedee) [inline CONTRIBUTING.md]: Explain why upstream remote

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+git remote add upstream [upstream_url]
+```
+
+Example: for our first-contributions repository-
+
+```
+git remote add upstream https://github.com/techcombd/first-contributions.git
+```
+
+Check if the upstream remote added succesfully-
```
Comment (mahedee) [inline CONTRIBUTING.md]: Use following command to check if the upstream remote added succesfully

**Review by mahedee — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (ratanparai) [reply CONTRIBUTING.md]: `origin` and `upstream` are English words. They do not have any special meaning in Git. I could choose any name. I do not have to use `upstream` and `origin` names as explained.

File: CONTRIBUTING.md (hunk 3 above)
Comment (ratanparai) [reply CONTRIBUTING.md]: I have used commonly used language like [Configuring a remote fork](https://help.github.com/articles/configuring-a-remote-for-a-fork/) or [git forks and upstream](https://www.atlassian.com/git/articles/git-forks-and-upstreams). Check there is no `run the following command` or anything like that. 

I think using those type of the same sentences may clutter the real contents.

**Comment by ratanparai** [comment]
@mahedee can you review the changes?If there isn't any big problem with the guide then I think we should merge it and any small fixes can be applied in future PRs.

**Review by mahedee — APPROVED** [review]

**Pull request merged**